src/main/java/pl/allegro/tech/build/axion/release/domain/VersionSorter.java

```diff
@@ -32,9 +40,11 @@ Result pickTaggedCommit(
         Set<Version> versions = new LinkedHashSet<>();
         LinkedHashMap<Version, Boolean> isVersionNextVersion = new LinkedHashMap<>();
         LinkedHashMap<Version, TagsOnCommit> versionToCommit = new LinkedHashMap<>();
+        boolean tagsFound = false;
 
         for (TagsOnCommit tagsEntry : taggedCommits.getCommits()) {
             List<String> tags = tagsEntry.getTags();
+            tagsFound = tagsFound || !tags.isEmpty();
```

Comment:
When this will happen? Since we are iterating through `taggedCommits` I would assume that each commit has at least 1 tag otherwise it would NOT be `tagged` 🤔

Reply:
yes, I was thinking about it, but either we'll have to enable debug everywhere, which will probably clutter the logs a lot, but maybe that's ok? or we'll have to indicate for which class we want to change the logging level, which is also probably doable, but maybe less user friendly? What do you think?

Reply:
I think that currently in the code there is actually no way for this class to have an empty tag collection, but there is also nothing in this class that would prevent the creation of such an instance. This class even has a special method for such an occasion ;) TagsOnCommit.empty() which creates an instance with an empty list of tags.
This code, I hope, will work correctly in both cases:
* if taggedCommits.getCommits() is empty tagsFound==false
* if the list of tags in taggedCommits.getCommits() is empty then tagsFound==false
The advantage is that we don't have to wonder if someone really thought it was a great idea to pass an empty list to the TagsOnCommit constructor.
You can also set a contract for TagsOnCommit, the list of tags cannot be and then we throw an exception otherwise. Then you can easily simplify this code as you say.
But I'm not insisting, if you feel that this is an unnecessary complication then I'll let you convince me.